.github/workflows/flutter.yaml

```
@@ -20,23 +20,33 @@ jobs:
         with:
           channel: 'stable'
       
+      - name: Check flutter sdk version
+        if: steps.check_files.outputs.files_exists == 'true'
+        run: flutter --version
+      
       - name: Get dependencies
         if: steps.check_files.outputs.files_exists == 'true'
         run: flutter pub get
+      
+      - name: Setup Dart Code Metrics
+        if: steps.check_files.outputs.files_exists == 'true'
+        run: dart pub add --dev dart_code_metrics
 
-      - name: Run static code analysis
+      - name: Dart Code Metrics analyze
         if: steps.check_files.outputs.files_exists == 'true'
-        uses: invertase/github-action-dart-analyzer@v1
-        with:
-          fatal-infos: true
+        run: dart run dart_code_metrics:metrics analyze lib --fatal-warnings --fatal-performance --fatal-style
+      
+      - name: Dart Code Metrics check-unnecessary-nullable
+        if: steps.check_files.outputs.files_exists == 'true'
+        run: dart run dart_code_metrics:metrics check-unnecessary-nullable --fatal-warnings --fatal-performance --fatal-style
 
-      - name: Dart Code Metrics
+      - name: Dart Code Metrics check-unused-files
         if: steps.check_files.outputs.files_exists == 'true'
-        uses: dart-code-checker/dart-code-metrics-action@v2
-        with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          fatal_warnings: true
-          fatal_style: true
+        run: dart run dart_code_metrics:metrics check-unused-files --fatal-warnings --fatal-performance --fatal-style
```

Nit: use line breaks for readability:
```suggestion
        run: dart run dart_code_metrics:metrics check-unused-files \
                --fatal-warnings \
                --fatal-performance \
                --fatal-style
```